packages/story-editor/src/app/font/fontProvider.js

```diff
@@ -51,73 +45,51 @@ function FontProvider({ children }) {
 
   const fonts = queriedFonts.length > 0 ? queriedFonts : curatedFonts;
 
-  useEffect(() => {
-    let mounted = true;
-    // Don't load custom fonts if not enabled.
-    // TODO: Don't load custom fonts when not requested by FontPicker.
-    if (!customFonts && getFonts && isCustomFontsEnabled) {
-      try {
-        (async () => {
-          const _customFonts = await getFonts({
-            service: 'custom',
-          });
-
-          if (!mounted) {
-            return;
-          }
-
-          const formattedFonts = _customFonts.map((font) => ({
-            id: font.family,
-            name: font.family,
-            value: font.family,
-            ...font,
-          }));
+  const getCustomFonts = useCallback(async () => {
+    if (customFonts || !getFonts || !isCustomFontsEnabled) {
```

Comment:
The `getFonts` checks here and below might not be needed anymore. I now recall that I originally added it because of `storyPageToDataUrl`, which uses `FontProvider` to load fonts, but doesn't need custom fonts or curated fonts.

So with this effect gone, this safeguard would be not needed anymore.

Reply:
So if `getFonts` is not provided, we will not render `FontPicker` component dropdown at all?

Reply:
Like this https://github.com/GoogleForCreators/web-stories-wp/pull/10643/commits/a27975166e512bd5494e5876fb28c60b1581811c

Reply:
No I meant

```diff
-if (customFonts || !getFonts || !isCustomFontsEnabled) {
+if (customFonts || !isCustomFontsEnabled) {
```

Reply:
I meant just removing the `|| !getFonts` check but I suppose it makes sense like this 👍

Reply:
https://github.com/GoogleForCreators/web-stories-wp/commit/a27975166e512bd5494e5876fb28c60b1581811c looks wrong because if we remove `getFonts` condition, the standalone editor will throw an error when `getFonts` isn't provided. To prevent this I had added `fonts.length > 0` so that `FontPicker` doesn't get rendered and `getFonts` is never called, but actually, that was a mistake because `fonts` will not have any value until `FontPicker` is rendered so font picked will not render in any case. Reverting the commit changes.